core/state/trie_prefetcher.go

```diff
@@ -214,8 +281,23 @@ func (p *triePrefetcher) trie(root common.Hash) Trie {
 // used marks a batch of state items used to allow creating statistics as to
 // how useful or wasteful the prefetcher is.
 func (p *triePrefetcher) used(root common.Hash, used [][]byte) {
-	if fetcher := p.fetchers[root]; fetcher != nil {
-		fetcher.used = used
+	if !metrics.EnabledExpensive {
```

Comment:
why does metrics config affect the logic

Reply:
`used` is an expensive metrics, not needed if EnabledExpensive is not enabled.